File: Sample.java
//****BUY AND SELL STOCK-2****
//Time complexity: o(n);
//Space complexity:o(1);

class Solution {
    public int maxProfit(int[] prices) {
        int slow=0;
        int fast=1;
        int max=0;
        int profit=0;
        while(fast<prices.length)
        {
            if(prices[slow]>prices[fast])
            {
                //No profit is generated
                slow++;
                fast++;
            }
            
            else if(prices[slow]<=prices[fast])
            {
                profit+=prices[fast]-prices[slow];
                
                slow++;
                fast++;
            }
        }
        return profit;
    }
}

File: sample.swift
// Time Complexity : 0(n)
// Space Complexity : 0(1)
// Did this code successfully run on Leetcode :
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach

    func maxProfit(_ prices: [Int]) -> Int {
        if prices.count == 1 {
            return 0
        }
        var profit = 0
        var msp = prices[0]
        for i in 1..<prices.count {
            msp = min(msp,prices[i])
            if msp < prices[i] {
                profit += prices[i] - msp
                msp = prices[i]
            }
        }
        return profit
    }